tools/alignment_tools.py

```diff
@@ -0,0 +1,186 @@
+''' Tools for manipulating the alignments seralized file '''
+
+#TODO change format
+#TODO merge alignments
+
+import argparse
+import datetime
+import os
+import sys
+import json
+import pickle
+
+class AlignmentData(object):
+    ''' Class to hold the alignment data '''
+
+    def __init__(self, alignments_dir):
+        self.extensions = ['json', 'p', 'yaml', 'yml']
+        self.alignments_file = None
+        self.alignments_format = None
+        self.alignments = None
+
+        self.load_alignments(alignments_dir)
+
+    def alignments_file_exists(self, alignments_base, extension):
+        ''' return 1 if a file is found '''
+        alignments_file = '{}.{}'.format(alignments_base, extension)
+        if os.path.isfile(alignments_file):
+            self.alignments_file = alignments_file
+            self.alignments_format = extension
+            return 1
+        return 0
+
+    def check_alignments_file(self, alignments_base):
+        ''' Checks that there is just one alignment file '''
+        file_count = 0
+        for extension in self.extensions:
+            file_count += self.alignments_file_exists(alignments_base, extension)
+        if file_count != 1:
+            raise ValueError ('1 alignments file should exist, but {} were found'.format(file_count))
+
+    def read_alignments(self):
+        ''' Read the alignments data from the correct format '''
+        print('Loading {} alignments from {}'.format(self.alignments_format, self.alignments_file))
+        alignments = open(self.alignments_file, 'r')
+        if self.alignments_format == 'json':
+            return json.loads(alignments.read())
+        elif self.alignments_format == 'p':
+            return pickle.loads(alignments.read())
+        elif self.alignments_format in ('yaml', 'yml'):
+            try:
+                import yaml
+            except ImportError:
+                print("You must have PyYAML installed to use YAMLSerializer")
+            return yaml.load(alignments.read())
+        alignments.close()
```

Comment:
The file will never be closed if the method returns somewhere within the if-elif-elif loop.

You need to use a `with open() as f:` block instead of the way you're doind it now.

Reply:
amending